misk/src/main/kotlin/misk/hash/ConsistentHashRing.kt

```diff
@@ -0,0 +1,44 @@
+package misk.hash
+
+import com.google.common.hash.HashFunction
+import com.google.common.hash.Hashing
+
+/** A [ConsistentHashRing] maps strings to values based on a consistent hash */
+class ConsistentHashRing<T>(
+  members: Map<String, T>,
+  private val hashFn: HashFunction = Hashing.murmur3_32(),
```

Comment:
Off topic: any idea whether this hash function is easy to get on Go? We don’t need cryptographic hashing and I suspect it’d be handy in future if we had the same hash rings on all of our platforms

Reply:
Yep: https://github.com/spaolacci/murmur3